### PR TITLE
simplify SDR iterate

### PR DESCRIPTION
**Pull request by Borda**
simplify the _iterate function in SDRegistration to be easier to read

### CODE REVIEW

**Comment by Borda** [comment]
I think that it would be nice to make some more code cleaning with a help of an automatic tool...

**Comment by skoudoro** [comment]
Hi @Borda, Thank you really much for this! 

I will get back to you ASAP concerning this big PR.

Concerning an automatic tool for checking code quality, I would like to have @Garyfallidis and @arokem opinion.

**Comment by Borda** [comment]
Btw, should the code also be by PEP8, because I saw many incompatibilities there...

**Comment by arokem** [comment]
We'd be happy to get PRs fixing up PEP8 compatibility, but please do it for
only a few files in each PR.

**Comment by skoudoro** [comment]
I agree with @arokem, it will be good if you can split this PR into 2 PRs at least: SDR simplification and code quality

**Comment by Borda** [comment]
@arokem @skoudoro Ok, see #1471

File: dipy/align/imwarp.py [hunk 1]
```diff
@@ -1264,6 +1255,38 @@ def _iterate(self):
...
+            self.callback(self, RegistrationStages.ITER_END)
+        self.metric.free_iteration()
+
+        return der
+
+    def __set_no_boundary_displacement(self, step):
+        """ set zero displacements at the boundary
+
+        :param ndarray step:
+        :return ndarray:
```
Comment (skoudoro) [inline dipy/align/imwarp.py]: Can you update the docstring by using  [Numpy documentation standard](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)

File: dipy/align/imwarp.py [hunk 2]
```diff
@@ -1264,6 +1255,38 @@ def _iterate(self):
...
+            step[:, :, 0, ...] = 0
+            step[:, :, -1, ...] = 0
+        return step
+
+    def __invert_models(self, current_disp_world2grid, current_disp_spacing):
+        """ converting static - moving models in both direction
+
+        :param current_disp_world2grid:
+        :param current_disp_spacing:
+        """
```
Comment (skoudoro) [inline dipy/align/imwarp.py]: same as above

**Comment by skoudoro** [comment]
It seems ready to go! 

I will merge it tomorrow. if there is no objection.

**Comment by Borda** [comment]
now in #1475 since somehow after rebase it was automatically closed

**Pull request closed without merging**